src/traces/sunburst/plot.js

```diff
@@ -471,235 +483,111 @@ function partition(entry) {
-        // would likely need to change pie to use dragElement instead of straight
-        // mapbox event binding. Or perhaps better, make a simple wrapper with the
-        // right mousedown, mousemove, and mouseup handlers just for a left/right click
-        // mapbox would use this too.
-        var fullLayoutNow = gd._fullLayout;
-        var traceNow = gd._fullData[trace.index];
-
-        var clickVal = Events.triggerHandler(gd, 'plotly_sunburstclick', {
-            points: [helpers.makeEventData(pt, traceNow)],
-            event: d3.event
-        });
-
-        // 'regular' click event when sunburstclick is disabled or when
-        // clikcin on leaves or the hierarchy root
-        if(clickVal === false || helpers.isLeaf(pt) || helpers.isHierachyRoot(pt)) {
-            if(fullLayoutNow.hovermode) {
-                gd._hoverdata = [helpers.makeEventData(pt, traceNow)];
-                Fx.click(gd, d3.event);
+        for(var q = 0; q < cd.length; q++) {
+            if(cd[q].label === id) {
+                return id;
             }
-            return;
         }
+        return '';
+    };
 
-        // skip if triggered from dragging a nearby cartesian subplot
-        if(gd._dragging) return;
-
-        // skip during transitions, to avoid potential bugs
-        // we could remove this check later
-        if(gd._transitioning) return;
-
-        // store 'old' level in guiEdit stash, so that subsequent Plotly.react
-        // calls with the same uirevision can start from the same entry
-        Registry.call('_storeDirectGUIEdit', traceNow, fullLayoutNow._tracePreGUI[traceNow.uid], {level: traceNow.level});
-
-        var hierarchy = cd0.hierarchy;
-        var id = helpers.getPtId(pt);
-        var nextEntry = helpers.isEntry(pt) ?
-            findEntryWithChild(hierarchy, id) :
-            findEntryWithLevel(hierarchy, id);
+    var getVal = function(d) {
+        if(d.hasOwnProperty('hierarchy')) return d.hierarchy.value;
+        return d.hasOwnProperty('v') ? d.v : d.value;
+    };
 
-        var frame = {
-            data: [{level: helpers.getPtId(nextEntry)}],
-            traces: [trace.index]
-        };
+    var calcPercent = function() {
+        var result = (trace.branchvalues ? cdi.v : cdi.value) / getVal(ref);
+        return isFinite(result) ? result : 1;
+    };
 
-        var animOpts = {
-            frame: {
-                redraw: false,
-                duration: constants.CLICK_TRANSITION_TIME
-            },
-            transition: {
-                duration: constants.CLICK_TRANSITION_TIME,
-                easing: constants.CLICK_TRANSITION_EASING
-            },
-            mode: 'immediate',
-            fromcurrent: true
-        };
-
-        Fx.loneUnhover(fullLayoutNow._hoverlayer.node());
-        Registry.call('animate', gd, frame, animOpts);
-    });
-}
+    if(trace.type === 'treemap' && helpers.isHeader(pt, trace)) {
```

Comment:
I don't think we need this `if` block. Header nodes don't appear to call `formatSliceLabel` from:

https://github.com/plotly/plotly.js/blob/217c0f43020f9f13d4361aadc6cd2d1a93c020ae/src/traces/treemap/draw_descendants.js#L167

Reply:
... with the `formatSliceLabel` below:

https://github.com/plotly/plotly.js/blob/217c0f43020f9f13d4361aadc6cd2d1a93c020ae/src/traces/treemap/draw_descendants.js#L167-L169

Reply:
Good call. Moved to `treemap/draw_descendants.js` in https://github.com/plotly/plotly.js/pull/4185/commits/8416883be97b77bdf685ade8790544e4d3b0fa8a.